Init

eldadfux opened the pull request.

eldadfux committed:
Init

eldadfux committed:
Removed vendor dir

eldadfux committed:
Format

eldadfux committed:
Add GitHub issue templates and CI workflows

eldadfux committed:
Fix PHPUnit test configuration path

Meldiron commented on .github/ISSUE_TEMPLATE/bug.yaml:
```diff
@@ -0,0 +1,81 @@
+  - type: dropdown
+    id: appwrite-version
+    attributes:
+      label: "🎲 Utopia Storage Version"
+      description: "What version of Utopia Storage are you running?"
```
```suggestion
      label: "🎲 Utopia Auth Version"
      description: "What version of Utopia Auth are you running?"
```

Meldiron commented on LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 Utopia
```
```suggestion
Copyright (c) 2025 Utopia
```

Meldiron commented on README.md:
````diff
@@ -0,0 +1,114 @@
+$hash = $scrypt->hash('password');
+```
+
+### Authentication Proofs
+
+The library also supports various authentication proofs for implementing secure authentication flows:
+
+```php
+<?php
+
+use Utopia\Auth\Proofs;
+
+// Initialize proofs
+$proofs = new Proofs();
+
+// Add your authentication proof logic here
+```
````
Example doesnt give too much value

Meldiron commented on .github/workflows/linter.yml:
```diff
@@ -0,0 +1,20 @@
+name: "Linter"
```
Lets also add static code analysis.

- https://github.com/utopia-php/logger/blob/main/.github/workflows/code-ql-analysis.yml#L1
- https://github.com/utopia-php/logger/blob/main/composer.json#L8
- https://github.com/utopia-php/logger/blob/main/composer.json#L34

Meldiron commented on src/Auth/Algorithms/Sha.php:
```diff
@@ -0,0 +1,82 @@
+
+class Sha extends Algorithm
+{
+    public const SHA1 = 'sha1';
+
+    public const SHA224 = 'sha224';
+
+    public const SHA256 = 'sha256';
+
+    public const SHA384 = 'sha384';
+
+    public const SHA512 = 'sha512';
+
+    public const SHA3_224 = 'sha3-224';
+
+    public const SHA3_256 = 'sha3-256';
+
+    public const SHA3_384 = 'sha3-384';
+
+    public const SHA3_512 = 'sha3-512';
```
```suggestion
    public const SHA1 = 'sha1';
    public const SHA224 = 'sha224';
    public const SHA256 = 'sha256';
    public const SHA384 = 'sha384';
    public const SHA512 = 'sha512';
    public const SHA3_224 = 'sha3-224';
    public const SHA3_256 = 'sha3-256';
    public const SHA3_384 = 'sha3-384';
    public const SHA3_512 = 'sha3-512';
```

Meldiron commented on src/Auth/Proofs/Code.php:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @throws \Exception
+     */
+    public function __construct(int $length = 6)
```
Might be good to have setLength

Meldiron commented on src/Auth/Proofs/Code.php:
```diff
@@ -0,0 +1,57 @@
+     */
+    public function hash(string $proof): string
+    {
+        return \hash('sha256', $proof);
```
This should be SHA256 Algorithm I would say

Meldiron replied:
Also, why enforce? Can we accept it as parameter in constructor - any algorithm, SHA256 being defaunt?

eldadfux replied:
What do we do today on Appwrite?

eldadfux committed:
Update .github/ISSUE_TEMPLATE/bug.yaml

Co-authored-by: Matej Bačo <matejbaco2000@gmail.com>

Meldiron commented on src/Auth/Proofs/Phrase.php:
```diff
@@ -0,0 +1,57 @@
+     */
+    public function __construct()
+    {
+        parent::__construct('argon2', ['argon2' => new Argon2()]);
```
Good default, but I would accept param to customize

eldadfux replied:
Do we use any hashing with this today on Appwrite?

eldadfux committed:
Update LICENSE

Co-authored-by: Matej Bačo <matejbaco2000@gmail.com>

Meldiron commented on src/Auth/Proofs/Phrase.php:
```diff
@@ -0,0 +1,57 @@
+     */
+    public function hash(string $proof): string
+    {
+        return password_hash($proof, PASSWORD_DEFAULT);
```
This is not using algorithm class; if ago was md5, this would still be using argon

eldadfux committed:
Update src/Auth/Algorithms/Sha.php

Co-authored-by: Matej Bačo <matejbaco2000@gmail.com>

Meldiron commented on src/Auth/Proofs/Phrase.php:
```diff
@@ -0,0 +1,57 @@
+     */
+    public function verify(string $proof, string $hash): bool
+    {
+        return password_verify($proof, $hash);
```
This is not using algorithm class; if ago was md5, this would still be using argon

Meldiron commented on src/Auth/Proofs/Token.php:
```diff
@@ -0,0 +1,54 @@
+     */
+    public function hash(string $proof): string
+    {
+        return \hash('sha256', $proof);
```
Same comment as before; we might want to make algo customizable

Meldiron commented on tests/Auth/Algorithms/ShaTest.php:
```diff
@@ -0,0 +1,42 @@
+use PHPUnit\Framework\TestCase;
+use Utopia\Auth\Algorithms\Sha;
+
+class ShaTest extends TestCase
```
We are missing tests for:
- PHPass
- md5
- argon

eldadfux committed:
Refactor Proof classes to use a more flexible algorithm strategy

eldadfux committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into dev

eldadfux committed:
format

eldadfux committed:
Update Proof classes with algorithm improvements and test adjustments

- Set default algorithm in Password class when initializing
- Update tests to use new algorithm methods and default settings
- Modify token generation to explicitly set SHA algorithm
- Adjust test assertions to match new default hashing methods

eldadfux committed:
Format

eldadfux committed:
Enhance Proof classes with dynamic generation and improved flexibility

- Update Proof classes to generate random values without input parameter
- Add password generation with configurable length and charset
- Modify README.md with comprehensive usage examples for different proof types
- Update tests to reflect new generation and configuration methods
- Improve code flexibility and security for authentication proofs

eldadfux committed:
Added missing tests

eldadfux committed:
code quality fixes

eldadfux committed:
Fixed phpstan errors

eldadfux committed:
Format

Meldiron commented on phpstan.neon:
Lets remove the file, not needed anymore

eldadfux replied:
Removed

eldadfux committed:
Not needed

Meldiron commented on src/Auth/Proofs/Password.php:
```diff
@@ -0,0 +1,190 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function hash(string $proof): string
+    {
+        return $this->algorithm->hash($proof);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function verify(string $proof, string $hash): bool
+    {
+        return $this->algorithm->verify($proof, $hash);
+    }
```
Those can be set on Proof abstract class, like setAdapter is.

eldadfux committed:
Refactor Proof classes to centralize hash and verify methods

Move common hash and verify implementations from individual Proof subclasses to the abstract Proof base class, reducing code duplication and simplifying the class hierarchy
